test/other-issues/sqlite-relations-too-many-parameters/entity/Category.ts

```diff
@@ -5,14 +5,14 @@ import {PrimaryColumn} from "../../../../src/decorator/columns/PrimaryColumn";
 export class Category {
 
     @PrimaryColumn()
-    id: string;
+    id: number;
```

Comment:
This one is a bit tricky. If ids are strings they're passed in query parameters, if numbers in query text. 1200 parameters are more than standard sqlite can handle. We should either accept that in some rare cases when PK is string there will be errors on sqlite or remove optimizations on how entities are loaded.
https://github.com/typeorm/typeorm/blob/20ff8c60a50df13517ed13451c0bba1c10d4f1c8/src/query-builder/SelectQueryBuilder.ts#L2071-L2094

We could also escape string ids and pass them as query text, but it would be harder to implement and more error prone.

Reply:
yeah we need to resolve this one. Don't think we should change this test just to make things work... there is a real issue in there.

Reply:
Ok, so we probably will have to remove that many-to-many optimization on query builder. I will look into that after you push code you mentioned earlier(if it won't fix the problem ofc.).